test/generation/test_gen.py

```diff
@@ -227,6 +213,15 @@ def test_gen_candidates_scipy_warns_opt_failure(self):
         )
         self.assertTrue(expected_warning_raised)
 
+    def test_gen_candidates_scipy_no_maxiter_warning(self):
+        # Check that no warnings are raised on hitting maxiter.
```

Comment:
We should probably log this? Could be at the info or maybe even the warning logging level.